### PR TITLE
Fixes issue #4150: Session management

### PR DESCRIPTION
**Pull request by flovilmart**

### CODE REVIEW

**Comment by miguel-s** [comment]
Hi @flovilmart 

We have been testing this fix on our own server and have found two things:
1. it successfully blocks updating of sessions (with PUT)
2. it does not prevent creating new sessions with a different user object (with POST). As far as we can see the user pointer saved in sessionData gets accidentally overwritten with the user pointer in the request data in the for-loop on [line 673](https://github.com/parse-community/parse-server/pull/4152/files#diff-3646535e0655c134862bb00401c8097cR673). To fix this issue we have added the user key to the if condition on [line 674](https://github.com/parse-community/parse-server/pull/4152/files#diff-3646535e0655c134862bb00401c8097cR674) like this `if (key === 'objectId' || key === 'user')`.

With your fix and this change we are unable to reproduce issue #4150.

**Comment by flovilmart** [comment]
@miguel-s thanks for debunking it, do you wanna make a PR on my PR with the additional tests so we can merge it?

**Comment by flovilmart** [comment]
Thanks @miguel-s , I'll let @acinader give the final review OK :)

**Comment by flovilmart** [comment]
@nathanrolnik, can you have a look?

File: spec/rest.spec.js [hunk 1]
```diff
@@ -498,6 +499,99 @@ describe('rest create', () => {
...
+        const user = new Parse.User();
+        user.id = currentUser.id;
+        return rp.put({
+          headers,
+          url: 'http://localhost:8378/1/sessions/' + sessionId,
+          json: {
+            user: Parse._encode(user)
+          }
+        })
+      }).then(done).catch(done.fail);
```
Comment (acinader) [inline spec/rest.spec.js]: you don't need the catch after the done since you catch up a level....

Comment (flovilmart) [reply spec/rest.spec.js]: right it was just to be safe, I can remove.

**Pull request merged**